spring-cloud-commons/src/main/java/org/springframework/cloud/client/DefaultServiceInstance.java

```diff
@@ -45,22 +47,41 @@ public class DefaultServiceInstance implements ServiceInstance {
 
 	private URI uri;
 
+	private int weight;
```

Comment:
In keeping with the response to https://github.com/spring-cloud/spring-cloud-commons/issues/1113, in order to ensure that these values (weight and current weight) are being passed back and forth between the service registry and the LB clients, they should be kept and updated in the metadata.

Reply:
In my original thought, weight must be by `final` to ensure the weight of entry is immutable, but it's not because I chose to keep the style consistent with the other fields. I will response more in the issue.

Reply:
Please provide your response on this topic either here or in the issue so that we can move forward with this.

Reply:
Please do not resolve conversations on the PR until either a change is made or an agreement is reached in the discussion in the comments.

Reply:
Sorry, I now see you've added that below.